### PR TITLE
Refactor NEGO NLA and redirection to use opaque handles

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/core/nla.c [hunk 1]
```diff
@@ -99,6 +99,53 @@
...
+	LPTSTR SspiModule;
+	char* SamFile;
+	rdpSettings* settings;
+	rdpTransport* transport;
+	UINT32 cbMaxToken;
+#if defined(UNICODE)
+	SEC_WCHAR* packageName;
+#else
+	SEC_CHAR* packageName;
+#endif
```
Comment (informatimago) [inline libfreerdp/core/nla.c]: This is what TCHAR is for.

Comment (akallabeth) [reply libfreerdp/core/nla.c]: Correct, did not look at the content of the struct, just moved it

**Comment by akallabeth** [comment]
@mfleisz done :)

**Comment by mfleisz** [comment]
@akallabeth I think the correct data type (at least on Windows) should be ```SECURITY_PSTR```

**Comment by mfleisz** [comment]
@akallabeth Alright 👍

**Pull request merged**